kong/clustering/control_plane.lua

```diff
@@ -59,50 +59,30 @@ local REMOVED_FIELDS = require("kong.clustering.compat.removed_fields")
 local _log_prefix = "[clustering] "
 
 
+local plugins_list_to_map = clustering_utils.plugins_list_to_map
+
+
 local function handle_export_deflated_reconfigure_payload(self)
   local ok, p_err, err = pcall(self.export_deflated_reconfigure_payload, self)
   return ok, p_err or err
 end
 
 
-local function plugins_list_to_map(plugins_list)
-  local versions = {}
-  for _, plugin in ipairs(plugins_list) do
-    local name = plugin.name
-    local version = plugin.version
-    local major, minor = clustering_utils.extract_major_minor(plugin.version)
-
-    if major and minor then
-      versions[name] = {
-        major   = major,
-        minor   = minor,
-        version = version,
-      }
-
-    else
-      versions[name] = {}
-    end
-  end
-  return versions
-end
-
-
 local function is_timeout(err)
   return err and sub(err, -7) == "timeout"
 end
 
 
-function _M.new(parent)
+function _M.new(initer)
```

Comment:
"initer" is still a bundle with unrelated parameters. Could we just separately pass those arguments?
Wouldn't this be better?
```lua
    self.json_handler = require("kong.clustering.control_plane").new(self.conf, self.cert_digest)
    self.wrpc_handler = require("kong.clustering.wrpc_control_plane").new(self.conf, self.cert_digest)
```